### PR TITLE
Move Towards Mouse Example

### PR DESCRIPTION
**Pull request by ScriptLineStudios**
#69 - simple move towards mouse example. Should be easy to integrate with a future ```pygame.math.Vector2.move_towards``` function.

### CODE REVIEW

File: pgex/examples/move_towards_mouse/vector2.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        """
+        Returns the sine and cosine needed to move the vector towards the target at the given speed.
+        """
+
+        dy, dx = dst_y - self.y, dst_x - self.x
+        angle = math.atan2(dy, dx)
+
+        if dx != 0 and dy != 0:
+            return (math.cos(angle) * speed, math.sin(angle) * speed)
+        else:
```
Comment (pycoinfu) [inline pgex/examples/move_towards_mouse/vector2.py:29]: do we really need an `else` here 🤔

**Review by pycoinfu — APPROVED** [review]
lgtm
when 2.1.3 comes out, we can rewrite this to use `pygame.Vector2.move_towards`

File: pgex/examples/move_towards_mouse/vector2.py (hunk 1 above)
Comment (ScriptLineStudios) [reply pgex/examples/move_towards_mouse/vector2.py:29]: Running without leads to some jittering when the entity gets close to the target if you remove the smoothing

Comment (Matiiss) [reply pgex/examples/move_towards_mouse/vector2.py:29]: I think the point is that if previous `return` is reached then code after that won't be reached, the second `return` can be unindented and `else` removed

File: pgex/examples/move_towards_mouse/vector2.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        """
+        Returns the sine and cosine needed to move the vector towards the target at the given speed.
+        """
+
+        dy, dx = dst_y - self.y, dst_x - self.x
+        angle = math.atan2(dy, dx)
+
+        if dx != 0 and dy != 0:
+            return (math.cos(angle) * speed, math.sin(angle) * speed)
+        return (0, 0)
```
Comment (Matiiss) [inline pgex/examples/move_towards_mouse/vector2.py:27]: can be simplified to
```py
if dx and dy:
    return math.cos(angle) * speed, math.sin(angle) * speed
return 0, 0
```

**Review by Matiiss — COMMENTED** [review]
Otherwise looked good!

File: pgex/examples/move_towards_mouse/vector2.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    def move_towards(self, dst_x, dst_y, speed) -> Tuple[float, float]:
+        """
+        Returns the sine and cosine needed to move the vector towards the target at the given speed.
+        """
+
+        dy, dx = dst_y - self.y, dst_x - self.x
+        angle = math.atan2(dy, dx)
+
+        if dx != 0 and dy != 0:
```
Comment (Matiiss) [inline pgex/examples/move_towards_mouse/vector2.py:25]: Simplify to `if dx and dy:`

**Review by Matiiss — CHANGES_REQUESTED** [review]
Simplify to `if dx and dy:`

**Review by Matiiss — APPROVED** [review]
LGTM!

**Pull request merged**